tests/security/selinux/semanage_boolean.pm

```diff
@@ -61,12 +61,10 @@ sub run {
     validate_script_output(
         "semanage boolean -l -C",
         sub {
-            m/
-            SELinux.*boolean.*State.*Default.*Description.*
-            ${test_boolean}.*(on.*,.*on).*Allow.*to.*
-            selinuxuser_execmod.*(on.*,.*on).*Allow.*to.*/sx
+            m/(?=.*SELinux\s+boolean\s+State\s+Default\s+Description)(?=.*${test_boolean}\s+\(on\s+,\s+on\))(?=.*selinuxuser_execmod\s+\(on\s+,\s+on\))/s;
```

Comment:
I think [Positive Lookahead](https://www.regular-expressions.info/lookaround.html) does not work as you want.
And I'm not sure if the output must have specific order. Maybe only `\A(?=.*SELinux...)`

http://10.67.17.201/tests/1347#step/semanage_boolean/30

Reply:
The output has different order between sle12 and sle15.  i.e. it outputs " A B C" on one sle version. It outputs "B A C" on other version. The Positive Lookaheads can work like "and". The regex in the change means the output must have A  and B and C regardless of order. 
I was interested in this usage so did some research. See this for detail http://www.rexegg.com/regex-lookarounds.html
One thing that article did not mention is that the Lookaheads should not conflict each other. Any should cover the others.

Reply:
If regardless of the order then I don't understand the purpose of Lookahead.

Both versions are passing
http://10.67.17.201/tests/1347#step/semanage_boolean/30
http://10.67.17.201/tests/1348#step/semanage_boolean/30

Reply:
The original regex is order-sensitive. As you can from screen shots of above two tests, orders are different. One (12sp5) would fail with original regex. 
The improved regex (the lookahead one) is order-insensitive. So both tests are passed. 
The improved regex only care about the content. It doesn't care about the order in which the content are listed.

Reply:
Ok, I'm just confused by the `Positive Lookahead` because this implies `(?=foo) Asserts that what immediately follows the current position in the string is foo`
I would just group the part of regex that can have any order with [(?:...)](http://www.rexegg.com/regex-disambiguation.html#noncap)

Reply:
Lookahead will reset it position after each look.  Take (?=.*aa)(?=.*bb)(.*cc) for example. It search for aa then reset its position.  It does the same for aa and cc. It returns true if aa bb cc are all in the string. The regex engine is still where it started. 
Put ".*"  in between "?=" and "aa" so "aa" doesn't have to immediately follow the current position.

Reply:
not sure that ?: would reset position.

Reply:
It would, I don;t mind if you are fine as it is.

Reply:
Thanks.